openslides_backend/models/check_models.py

```diff
@@ -0,0 +1,686 @@
+
+        if isinstance(field_type, RelationField):
+            foreign_id = model[field]
+            if foreign_id is None:
+                return
+
+            foreign_collection, foreign_field = self.get_to(field, collection)
+
+            self.check_reverse_relation(
+                collection,
+                model["id"],
+                model,
+                foreign_collection,
+                foreign_id,
+                foreign_field,
+                basemsg,
+                replacement,
+            )
+
+        elif isinstance(field_type, RelationListField):
+            foreign_ids = model[field]
+            if foreign_ids is None:
+                return
+
+            foreign_collection, foreign_field = self.get_to(field, collection)
+
+            for foreign_id in foreign_ids:
+                self.check_reverse_relation(
+                    collection,
+                    model["id"],
+                    model,
+                    foreign_collection,
+                    foreign_id,
+                    foreign_field,
+                    basemsg,
+                    replacement,
+                )
+
+        elif isinstance(field_type, GenericRelationField) and model[field] is not None:
+            foreign_collection, foreign_id = self.split_fqid(model[field])
+            foreign_field = self.get_to_generic_case(
+                collection, field, foreign_collection
+            )
+
+            self.check_reverse_relation(
+                collection,
+                model["id"],
+                model,
+                foreign_collection,
+                foreign_id,
+                foreign_field,
+                basemsg,
+                replacement,
+            )
+
+        elif (
+            isinstance(field_type, GenericRelationListField)
+            and model[field] is not None
+        ):
+            for fqid in model[field]:
+                foreign_collection, foreign_id = self.split_fqid(fqid)
+                foreign_field = self.get_to_generic_case(
+                    collection, field, foreign_collection
+                )
+
+                self.check_reverse_relation(
+                    collection,
+                    model["id"],
+                    model,
+                    foreign_collection,
+                    foreign_id,
+                    foreign_field,
+                    basemsg,
+                    replacement,
+                )
+
+    def get_to(self, field: str, collection: str) -> Tuple[str, Optional[str]]:
+        if self.is_structured_field(field):
+            field, _ = self.to_template_field(collection, field)
+
+        field_type = cast(BaseRelationField, self.models[collection]().get_field(field))
+        return (
+            field_type.get_target_collection().collection,
+            field_type.to.get(field_type.get_target_collection()),
+        )
+
+    def find_model(self, collection: str, id: int) -> Optional[Dict[str, Any]]:
```

Comment:
Maybe create a new issue: This might be not performant for many models of one collection -> One can create a dict mapping `id` to the model for quick access.

Reply:
Add an issue: https://github.com/OpenSlides/openslides-backend/issues/827